recipes-kernel/linux/linux-adi/0001-sc59x-fastautomdix-support.patch

```diff
@@ -0,0 +1,97 @@
+diff --git a/arch/arm/boot/dts/sc594-som-ezkit.dts b/arch/arm/boot/dts/sc594-som-ezkit.dts
+index 25acefc408c2..9c1ac91c47b5 100644
+--- a/arch/arm/boot/dts/sc594-som-ezkit.dts
++++ b/arch/arm/boot/dts/sc594-som-ezkit.dts
+@@ -51,24 +51,22 @@
+ 			              <&ssw0 4 GPIO_ACTIVE_LOW>;	/* SPI2D2_D3_EN */
+ 		};
+ 
+-/* The button GPIO conflicts with OSPI0_D4 for this
++/* The button GPIO conflicts with OSPI0_D4 for this */
+ 		button0: button@0 {
+ 			compatible = "adi,button-led";
+-			en-pins = <&ssw1 1 GPIO_ACTIVE_LOW>;	/* PUSHBUTTON_EN
+-			button_gpio = <48>; /* PD_00
+-			led_gpio = <35>; /* PC_03 / LED9 / LED1_KIT
++			en-pins = <&ssw1 1 GPIO_ACTIVE_LOW>;/* PUSHBUTTON_EN */
++			button_gpio = <50>; /* PD_00 */
++			led_gpio = <35>; /* PC_03 / LED9 / LED1_KIT */
+ 		};
+-*/
+ 
+ /* This also appears to interfere with the OSPI somehow...
+-   possible trace noise/interference?
++/*  possible trace noise/interference? */
+ 		button1: button@1 {
+ 			compatible = "adi,button-led";
+-			en-pins = <&ssw1 1 GPIO_ACTIVE_HIGH>; /* PUSHBUTTON_EN
+-			button_gpio = <112>; /* PH_00
+-			led_gpio = <34>; /* PC_02 / LED8 / LED2_KIT
++			en-pins = <&ssw1 1 GPIO_ACTIVE_HIGH>; /* PUSHBUTTON_EN */
++			button_gpio = <112>; /* PH_00 */
++			led_gpio = <34>; /* PC_02 / LED8 / LED2_KIT */
+ 		};
+-*/
+ 
+ 		core1-rproc@0x3108c000 {
+ 			compatible = "adi,remoteproc";
+diff --git a/drivers/net/phy/dp83867.c b/drivers/net/phy/dp83867.c
+index 3aa1ae211f5f..d03db6405499 100644
+--- a/drivers/net/phy/dp83867.c
++++ b/drivers/net/phy/dp83867.c
+@@ -23,6 +23,8 @@
+ #define MII_DP83867_ISR		0x13
+ #define DP83867_CTRL		0x1f
+ #define DP83867_CFG3		0x1e
++#define DP83867_CFG3_FASTLINK   (BIT(15))
++#define DP83867_CFG3_FAST_AUT_MDIX  (BIT(8))
+ 
+ /* Extended Registers */
+ #define DP83867_CFG4            0x0031
+@@ -424,6 +426,10 @@ static int dp83867_config_init(struct phy_device *phydev)
+ 		phy_write(phydev, DP83867_CFG3, val);
+ #if (!defined(CONFIG_ARCH_SC58X)) && (!defined(CONFIG_ARCH_SC57X))
+ 	}
++         /*Fast Auto MDIX is enabled for faster link up*/
```

Comment:
Can you add all these changes only if the speed is 100 Mbps and auto-negotiation is disabled, because it is only applicable for that link?

Reply:
By default we are setting PHY interface speed to 100Mbps and Auto negotiation to Disable state - So this check may not be required

Reply:
Why is the default 100Mbps done as default? We are not going to run it in startup. So, you can keep the default as whatever it was (1000Mbps and auto-negotiation on). Later in the script, we can choose to run at 100Mbps auto-negotiation off. The link-down to link-up is only on applying push button (sleep mode described), which will only run at 100Mbps as it was at 100Mbps to begin with.

Reply:
Removed default speed and Auto negotiation setting to default. We can handle that change in script.